src/NextGen/DonationForm/resources/propTypes.ts

```diff
@@ -1,21 +1,117 @@
-import {Element, Field, Group, SelectOption} from '@givewp/forms/types';
-import {UseFormRegisterReturn} from 'react-hook-form';
-import {FC} from 'react';
+import {Element, Field, FormDesign, Gateway, Group, Section as SectionType, SelectOption} from '@givewp/forms/types';
+import {FieldErrors, UseFormRegisterReturn} from 'react-hook-form';
+import {FC, FormHTMLAttributes, ReactNode} from 'react';
 
 export interface FieldProps extends Field {
     inputProps: UseFormRegisterReturn;
-    Label: FC;
-    ErrorMessage: FC;
+    Label: FC<FieldLabelProps | {}>;
+    ErrorMessage: FC<FieldErrors | {}>;
 }
 
+export interface GatewayFieldProps extends FieldProps {
+    gateways: Gateway[];
+}
+
+export type GatewayOptionProps = {
+    inputProps: UseFormRegisterReturn;
+    gateway: Gateway;
+    index: number;
+};
+
 export interface SelectFieldProps extends FieldProps {
     options: Array<SelectOption>;
 }
 
 export interface ElementProps extends Element {}
 
 export interface GroupProps extends Group {
-    fieldProps: {
-        [key: string]: FieldProps;
+    fields: {
+        [key: string]: FC<FieldProps>;
+    };
+}
+
+export interface HtmlProps extends ElementProps {
+    html: string;
+}
+
+export interface NameProps extends GroupProps {
+    fields: {
+        honorific?: FC<SelectFieldProps | {}>;
+        firstName: FC<FieldProps | {}>;
+        lastName: FC<FieldProps | {}>;
     };
 }
+
+export interface AmountProps extends FieldProps {
+    levels: Number[];
+    allowCustomAmount: boolean;
+}
+
+export interface ParagraphProps extends ElementProps {
+    content: string;
+}
+
+export interface SectionProps {
+    section: SectionType;
+    children: ReactNode;
+}
+
+export interface FormProps {
+    formProps: FormHTMLAttributes<unknown>;
+    children: ReactNode;
+    formError: string | null;
+    isSubmitting: boolean;
+}
+
+export interface FieldErrorProps {
+    error: string;
+}
+
+export interface FieldLabelProps {
+    label: string;
+    required: boolean;
+}
+
+export enum GoalType {
+    AMOUNT = 'amount',
+    DONATIONS = 'donations',
+    DONORS = 'donors',
+}
+
+export interface GoalProps {
+    currency: string;
+    type: GoalType;
+    currentAmount: number;
+    currentAmountFormatted: string;
+    targetAmount: number;
+    targetAmountFormatted: string;
+    goalLabel: string;
+    progressPercentage: number;
+    totalRevenue: number;
+    totalRevenueFormatted: string;
+    totalCountValue: number;
+    totalCountLabel: string;
+}
+
+export interface HeaderProps {
+    Title: FC<HeaderTitleProps | {}>;
+    Description: FC<HeaderDescriptionProps | {}>;
+    Goal: FC<GoalProps | {}>;
+}
+
+export interface HeaderDescriptionProps {
+    text: string;
+}
+
+export interface HeaderTitleProps {
+    text: string;
+}
+
+export interface NodeWrapperProps {
+    nodeType: keyof FormDesign & string;
+    // TODO: make type work as this keyof (keyof FormDesign) & string
```

Comment:
I recommend checking out the new `satisfies` operator in 4.9, in case that helps: https://devblogs.microsoft.com/typescript/announcing-typescript-4-9/#satisfies